integrated_channels/canvas/client.py

```diff
@@ -183,28 +236,33 @@ def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-
         pass
 
     # Private Methods
-
-    def _update_course_details(self, course_id, serialized_data):
+    def _update_course_details(self, course_id, course_details):
         """
         Update a course for image_url (and possibly other settings in future)
-        This is used only for settings that are not settable in the initial course creation
+        Also sets course to 'offer' state by sending 'course[event]=offer'
+
+        Arguments:
+          - course_details (dict): { 'image_url' } : optional used if present to course[image_url]
         """
+        response_code = None
+        response_text = None
+        # Providing the param `event` and setting it to `offer` is equivalent to publishing the course.
+        update_payload = {'course': {'event': 'offer'}}
         try:
             # there is no way to do this in a single request during create
             # https://canvas.instructure.com/doc/api/all_resources.html#method.courses.update
-            content_metadata_item = json.loads(serialized_data.decode('utf-8'))['course']
-            if "image_url" in content_metadata_item:
-                url = CanvasAPIClient.course_update_endpoint(
-                    self.enterprise_configuration.canvas_base_url,
-                    course_id,
-                )
-                # Providing the param `event` and setting it to `offer` is equivalent to publishing the course.
-                self._put(url, json.dumps({
-                    'course': {'image_url': content_metadata_item['image_url'], 'event': 'offer'}
-                }).encode('utf-8'))
+            url = CanvasAPIClient.course_update_endpoint(
+                self.enterprise_configuration.canvas_base_url,
+                course_id,
+            )
+            if "image_url" in course_details:
+                update_payload['course']['image_url'] = course_details['image_url']
+
+            response_code, response_text = self._put(url, json.dumps(update_payload).encode('utf-8'))
         except Exception:  # pylint: disable=broad-except
             # we do not want course image update to cause failures
```

Comment:
Please add an Error log here for alerting and tracking: I already have 2 different backlog tickets for bad course image URLs for other LMS systems, I don't want to add canvas to that list of problems I can't track.

Reply:
sure, it will now look like this on error

```
021-06-24 18:18:20,270 ERROR 55966 [integrated_channels.canvas.client] [user None] [ip None] client.py:278 - [integrated_channel:canvas][integrated_channel_enterprise_customer_uuid:873fec72-5a21-4f95-ac19-e3d67f9c4777][integrated_channel_lms_user:None][integrated_channel_course_key:edX+99887766]Failed to update details for course, edx_course_id=edX+99887766, canvas_course_id=6074. Details: a test only
```